src/lib/SessionStatus.svelte

```diff
@@ -29,26 +33,65 @@
 
   function checkSessionStatus() {
     var opIframe = document.getElementById('opIframe');
-    var message = `${sofClient.getClient().getState('clientId')} ${sofClient.getClient().getState('tokenResponse.session_state')}`;
+    var clientId = sofClient.getClient().getState('clientId');
+    var sessionId = sofClient.getClient().getState('tokenResponse.session_state');
+    var message = `${clientId} ${sessionId}`;
     // Send message to OP iframe
     opIframe?.contentWindow?.postMessage(message, OIDC_BASE);
   }
 
+  let backupInactivityTimer: NodeJS.Timeout | undefined;
+  function resetBackupInactivityTimer() {
+    if (backupInactivityTimer !== undefined) {
+        clearTimeout(backupInactivityTimer);
+      }
+      backupInactivityTimer = undefined;
+      backupInactivityTimer = setTimeout(() => goto('/logout'), BACKUP_INACTIVITY_TIMEOUT);
+  }
+  function onVisible_resetBackupInactivityTimer() {
+    if (document.visibilityState === "visible") {
+      onVisible_resetBackupInactivityTimer();
+    }
+  }
+
   async function processStatus(event: any) {
     if (event.origin === OIDC_BASE) {
       var data = event.data;
-      if (data === 'changed') {
-        try {
-          let res = await sofClient.getClient().refresh();
-          if (!sofClient?.getClient()?.getState("tokenResponse.access_token")) {
-            throw Error("Unable to refresh token after session state change. Logging out.");
+      if (validatingSession || (sessionCheckValid !== undefined && !sessionCheckValid)) {
+        // Don't check status if it's only been erroring, as in Safari w/o 3p cookies
+        // If check has been 'error' but is now something else, change state to valid
+        sessionCheckValid = (data !== 'error');
+        if (validatingSession && !sessionCheckValid) {
+          // Check has only errored, initialize timeout (or manual logout)
+          resetBackupInactivityTimer();
+          document.addEventListener('click', resetBackupInactivityTimer);
+          document.addEventListener('scroll', resetBackupInactivityTimer);
+          document.addEventListener('visibilitychange', onVisible_resetBackupInactivityTimer);
```

Comment:
Can `resetBackupInactivityTimer ` be called here instead?

Reply:
I moved the visibility handler to its own function since it needed an extra check to make sure the tab was becoming visible and not being switched away from, and I didn't think anonymous functions played as nicely with `document.removeEventListener` so I opted to name it. Let me know if that's misinformed though.